repo/manifest/manifest_manager_test.go

```diff
@@ -381,3 +382,63 @@ func TestManifestAutoCompaction(t *testing.T) {
 		require.NoError(t, mgr.b.Flush(ctx))
 	}
 }
+
+func TestManifestConfigureAutoCompaction(t *testing.T) {
+	ctx := testlogging.Context(t)
+	data := blobtesting.DataMap{}
+	item1 := map[string]int{"foo": 1, "bar": 2}
+	labels1 := map[string]string{"type": "item", "color": "red"}
+
+	mgr := newManagerForTesting(ctx, t, data)
+
+	compactionCount := 99
+
+	t.Setenv(autoCompactionEnvKey, strconv.Itoa(compactionCount))
+
+	for i := 0; i < compactionCount-1; i++ {
+		addAndVerify(ctx, t, mgr, labels1, item1)
+		require.NoError(t, mgr.Flush(ctx))
+		require.NoError(t, mgr.b.Flush(ctx))
+	}
+
+	// Should not trigger compaction
+	_, err := mgr.Find(ctx, labels1)
+	require.NoError(t, err)
+
+	foundContents := getManifestContentCount(ctx, t, mgr)
+
+	if got, want := foundContents, compactionCount-1; got != want {
+		t.Errorf("unexpected number of blocks: %v, want %v", got, want)
+	}
```

Comment:
can also use an assert here
```suggestion
	assert.Equal(t, compactionCount-1, foundContents, "unexpected number of manifest blobs")
```